src/gems/simulation/output_values.py

```diff
@@ -162,6 +137,18 @@ def _set(
 
             self._value[key] = value
 
+        def _set_basis_status(
+            self, timestep: Optional[int], scenario: Optional[int], status: str
+        ) -> None:
+            timestep = 0 if timestep is None else timestep
+            scenario = 0 if scenario is None else scenario
+            key = TimeScenarioIndex(timestep, scenario)
+            if key not in self._value:
```

Comment:
I think this has nothing to do here, this is done by the set method. Maybe the simplest is to add `self._basis_status[key] = status` to the `_set` method, in the logic that `_set` is responsible for setting all relevant attribute of the variable (then the `isMip` check should also be in `_set`)